src/api/MACOManagerAPI.ts

```diff
@@ -0,0 +1,225 @@
+   * @param  txOpts        Transaction options object conforming to `Tx` with signer, gas, and gasPrice data
+   * @return               Transaction hash
+   */
+  public async proposeAsync(macoManager: Address, txOpts: Tx): Promise<string> {
+    // Check that the price has indeed experienced a crossover
+    await this.assertPropose(macoManager);
+
+    // Check the current block.timestamp
+    const currentTimeStampInSeconds = new BigNumber(Date.now()).div(1000);
+    const lastProposalTimestamp = (await this.macoStrategyManager.lastProposalTimestamp(macoManager));
+
+    const moreThanTwelveHoursElapsed = currentTimeStampInSeconds.minus(lastProposalTimestamp).gt(TWELVE_HOURS);
+    const moreThanSixHoursElapsed = currentTimeStampInSeconds.minus(lastProposalTimestamp).gte(SIX_HOURS);
+
+    if (moreThanTwelveHoursElapsed) {
+      // If current timestamp > 12 hours since the last, call initialPropose
+      return await this.macoStrategyManager.initialPropose(macoManager, txOpts);
+    } else if (moreThanSixHoursElapsed) {
+      // If the current timestamp >6 and <12 hours since last call, call confirmPropose
+      return await this.macoStrategyManager.confirmPropose(macoManager, txOpts);
+    } else {
+      throw new Error('Less than 6 hours has elapsed since the last proposal timestamp');
+    }
+  }
+
+  /**
+   * Fetches the state variables of the Moving Average Crossover Manager contract.
+   *
+   * @param  macoManager         Address of the Moving Average Crossover Manager contract
+   * @return                     Object containing the state information related to the manager
+   */
+  public async getMovingAverageManagerDetailsAsync(macoManager: Address): Promise<MovingAverageManagerDetails> {
+    const [
+      auctionLibrary,
+      auctionTimeToPivot,
+      core,
+      lastProposalTimestamp,
+      movingAverageDays,
+      movingAveragePriceFeed,
+      rebalancingSetToken,
+    ] = await Promise.all([
+      this.macoStrategyManager.auctionLibrary(macoManager),
+      this.macoStrategyManager.auctionTimeToPivot(macoManager),
+      this.macoStrategyManager.coreAddress(macoManager),
+      this.macoStrategyManager.lastProposalTimestamp(macoManager),
+      this.macoStrategyManager.movingAverageDays(macoManager),
+      this.macoStrategyManager.movingAveragePriceFeed(macoManager),
+      this.macoStrategyManager.rebalancingSetTokenAddress(macoManager),
+    ]);
+
+    const [
+      riskAsset,
+      riskCollateral,
+      setTokenFactory,
+      stableAsset,
+      stableCollateral,
+    ] = await Promise.all([
+      this.macoStrategyManager.riskAssetAddress(macoManager),
+      this.macoStrategyManager.riskCollateralAddress(macoManager),
+      this.macoStrategyManager.setTokenFactory(macoManager),
+      this.macoStrategyManager.stableAssetAddress(macoManager),
+      this.macoStrategyManager.stableCollateralAddress(macoManager),
+    ]);
+
+    return {
```

Comment:
Can we add another lightweight method that's just the `lastProposalTimestamp`. All of these could be useful for the rebalancing manager app, maybe add this entire method to the system API? None of the properties change outside of `lastProposalTimestamp`. This is going to be called very often.